frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -1107,6 +1103,62 @@ mod tests {
 		})
 	}
 
+	#[test]
+	fn ocw_lock_prevents_overlapping_execution() {
+		use crate::mock::{System, BlockNumber};
+		use sp_runtime::offchain::storage_lock::{StorageLock, BlockAndTime};
+
+		{
+			// first, ensure that a successful execution releases the lock
+			let (mut ext, pool) = ExtBuilder::default().build_offchainify(0);
+			ext.execute_with(|| {
+				let guard = StorageValueRef::persistent(&OFFCHAIN_LOCK);
+				let last_block = StorageValueRef::persistent(OFFCHAIN_LAST_BLOCK);
+
+				roll_to(25);
+				assert!(MultiPhase::current_phase().is_unsigned());
+
+				// initially, the lock is not set.
+				assert!(matches!(guard.get::<bool>(), None));
+
+				// a successful a-z execution.
+				MultiPhase::offchain_worker(25);
+				assert_eq!(pool.read().transactions.len(), 1);
+
+				// afterwards, the lock is not set either..
+				assert!(matches!(guard.get::<bool>(), None));
+				assert!(matches!(last_block.get::<BlockNumber>(), Some(Some(25))));
+			});
+		}
+
+		{
+			// ensure that if the guard is in hold, a new execution is not allowed.
+			let (mut ext, pool) = ExtBuilder::default().build_offchainify(0);
```

Comment:
I still think this should be two separate tests; that way each can run in isolation.

```suggestion
		}
	}
	
	#[test]
	fn ocw_lock_prevents_new_execution_while_guard_held() {
		{
			// ensure that if the guard is in hold, a new execution is not allowed.
			let (mut ext, pool) = ExtBuilder::default().build_offchainify(0);
```